frontend/src/metabase/core/components/Button/Button.styled.tsx

```diff
@@ -11,6 +11,12 @@ export const ButtonRoot = styled.button<ButtonRootProps>`
   transition: all 200ms linear;
   flex-shrink: 0;
 
+  &:hover {
+    /* XXX: standardize button transition, it wouldn't make much sense to only extend this in a single place, since transition should be a part of the design system. */
+    transition-property: color, border-color, background-color, opacity,
```

Comment:
Use base values from https://github.com/metabase/metabase/blob/147e1c911cd91fffa3cc30a4cea50a14d3a65c46/frontend/src/metabase/css/components/buttons.css#L25
and add `opacity` and `visibility` to allow fading in and out buttons.


___
from @ranquild [[the original comment](https://github.com/metabase/metabase/pull/29569#discussion_r1165248821)]
> I'd rather not add `transition` for `visibility` to buttons by default because I don't think it's expected that buttons are hidden after some delay by default. Could you specify required transition properties for your apply button only?


I want to hear @mazameli 's opinion on this. I think we've had a couple of similar discussions about whether a certain style should be scoped to a specific UI components, or should that be baked into the code components.

From a design's point of view, I couldn't think of a way to want to only have a few buttons that would fade in and out while most of the other buttons instantly appear and disappear since that creates inconsistency in the design.

Let me know if fading in and out buttons should only be scoped to the **Apply** button specifically for this project.

Reply:
Making `visibility` work with a delay (by default) is something I would not expect as a developer. I see no reason why we should patch all buttons for the case of the apply button.